src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,92 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a deadline object.
+ * It has an end date and time.
+ */
+public class Deadline extends Task {
+    private final LocalDate deadlineDate;
+
+
+    private static String[] parseDeadline(String task) throws DukeException {

```

Comment:
I think this would be better placed in the Parser object! Or at least handled there in some way.

Reply:
Thanks for your input! I feel by putting it into the respective class, it will be more modularized and clearer what are the requirements for a deadline.